### ISSUE TITLE
`conf/workers-shared-extra.yaml.j2` is applied to monolith Complement runs

### ISSUE THREAD

**Issue by MadLittleMods** [matrix-org/synapse#13590]
[`docker/complement/conf/workers-shared-extra.yaml.j2`](https://github.com/matrix-org/synapse/blob/3dd175b628bab5638165f20de9eade36a4e88147/docker/complement/conf/workers-shared-extra.yaml.j2) seems to apply even when `SYNAPSE_COMPLEMENT_USE_WORKERS=true` isn't set. I assume they aren't supposed to stack like that for the monolith?

Noticed because I saw the `reject_limit` as `99999` when investigating https://github.com/matrix-org/synapse/pull/13541#discussion_r952064254

**Comment by DMRobertson** [comment]
Don't have much familiarity here. @reivilibre might be able to advise?

**Comment by H-Shay** [comment]
Marking this as occasional as it doesn't occur on all runs, and tolerable as it seems that there is a workaround.

**Comment by richvdh** [comment]
> I assume they aren't supposed to stack like that for the monolith?

No, I think this is expected behaviour. As the first two lines in the file say:

>  This file extends the default 'shared' configuration file (from the 'synapse-workers'
>  docker image) with Complement-specific  tweak.
  
I think it's just ended up with a bit of a misleading name, so the solution here is just to rename it?
